dsp/dsp.py

```diff
@@ -0,0 +1,590 @@
+        Dsp.to_orcid_window(self.driver)
+        self.assertIn("ORCID", TestSystem.title(self.driver))
+
+        OrcidWindow.fill_credentials(self.driver, USERNAME, PASSWORD)
+        OrcidWindow.to_origin_window(self.driver, wait=True)
+
+    def login_orcid_to_submit(self, repo):
+        self.login_orcid()
+        Dsp.show_mobile_nav(self.driver)
+        Dsp.drawer_to_submit(self.driver)
+        SubmitLandingPage.select_repo_by_id(self.driver, repo)
+
+
+class DspHydroshareTestSuite(DspTestSuite):
+    """DSP tests for Hydroshare repository"""
+
+    repo_name = "HydroShare"
+
+    @classmethod
+    def required_elements_template(self, auto_text):
+        basic_info = {
+            "Title": auto_text + " Title",
+            "Abstract": auto_text + " Abstract",
+            "Subjectkeywords": [auto_text + " SubjectKeywords"]
+        }
+        funding_agency = {
+            "Agencyname": auto_text + " Agencyname",
+        }
+
+        # created separately so that we can check individually if needed
+        required_elements = {
+            "BasicInformation": basic_info,
+            "Fundingagencyinformation": funding_agency,
+        }
+        return required_elements
+
+    def login_orcid_and_hs(self):
+        """Authenticate with orcid and then HS credentials"""
+        Dsp.show_mobile_nav(self.driver)
+        Dsp.drawer_to_submit(self.driver)
+        SubmitLandingPage.select_repo_by_id(self.driver, self.repo_name)
+
+        # new ORCID window
+        SubmitLandingPage.to_orcid_window(self.driver)
+        self.assertIn("ORCID", TestSystem.title(self.driver))
+
+        OrcidWindow.fill_credentials(self.driver, USERNAME, PASSWORD)
+        OrcidWindow.to_origin_window(self.driver)
+
+        # new HS auth window
+        SubmitLandingPage.to_repo_auth_window(self.driver)
+        self.assertIn(self.repo_name, TestSystem.title(self.driver))
+        HydroshareAuthWindow.authorize_repo(self.driver, HS_USERNAME, HS_PASSWORD)
+        HydroshareAuthWindow.to_origin_window(self.driver)
+
+    def login_and_autofill_hs_required(self, auto_text):
+        """A shortcut to fill required fields of HS submit page
+        So that additional non-required fields can easily be checked
+        """
+        self.login_orcid_and_hs()
+        template = self.required_elements_template(auto_text)
+        SubmitHydroshare.autofill_required_elements(self.driver, template)
+
+    def fill_ids_submit_and_check(self, sort_text, section, nth, dict):
+        """Fill additional fields of HS submit page based on 'data-id'
+        Then submit the form, search in 'My Submissions',
+        and check that all of the fields match what was entered
+        """
+        success_filling = SubmitHydroshare.fill_inputs_by_data_ids(self.driver, dict, section, nth)
+        self.assertTrue(success_filling)
+        SubmitHydroshare.finish_submission(self.driver)
+
+        # The page isn't sorted upon load
```

Comment:
Is this a bug that should be fixed instead of accounting for it in our automated tests?

Reply:
yep I have an open ticket for it here:
https://github.com/cznethub/dspfront/issues/53